presto-main/src/main/java/io/prestosql/spiller/SpillCipher.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spiller;
+
+import java.nio.ByteBuffer;
+
+public interface SpillCipher
+{
+    /**
+     * Encrypts the source data
+     */
+    byte[] encrypt(byte[] data);
+
+    /**
+     * Encrypts the contents of the input ByteBuffer into a new ByteBuffer instance. The returned instance will
+     * be at position 0 and ready to read.
+     */
+    ByteBuffer encrypt(ByteBuffer data);
+
+    /**
+     * Decrypts encrypted data
+     */
+    byte[] decrypt(byte[] encryptedData);
+
+    /**
+     * Decrypts the given {@link ByteBuffer} contents and returns them in a new {@link ByteBuffer}. The returned
+     * instance will be at position 0 and ready to read.
+     */
+    ByteBuffer decrypt(ByteBuffer encryptedData);
+
+    /**
+     * Destroy encryption key preventing future use. Implementations should allow this to be called multiple times
+     * without failing.
+     */
+    void destroy();
```

Comment:
Any reason not to implement `AutoCloseable` and have the signature here:
```
void close();
```
?

Reply:
I think `close()` is a little misleading about the nature of the method while `destroy()` is more clear about what it does, but I see the convenience of `SpillCipher` implementing `AutoCloseable` for the purposes of using it with `Closer`.

What do you think about `SpillCipher` extending `AutoCloseable` and adding:
```java
    @Override
    default void close()
    {
        destroy();
    }
```

Reply:
We're all pretty used to the misleading nature of `close`, so I would go with that over having two methods.  If you are using try finally, auto closable is really nice, but if you are only using `Closer` it doesn't really matter that much.

Reply:
Ok, will just use `close()` then.